apps/wallet/src/ui/app/hooks/useDeepBook.ts

```diff
@@ -105,11 +121,11 @@ async function getDeepbookPricesInUSD(coins: Coins[], deepBookClient: DeepBookCl
 }
 
 function useDeepbookPricesInUSD(coins: Coins[]) {
+	const deepBookClient = useDeepBookClient();
 	return useQuery({
-		queryKey: [DEEPBOOK_KEY, 'get-prices-usd', coins],
+		// eslint-disable-next-line @tanstack/query/exhaustive-deps
+		queryKey: [DEEPBOOK_KEY, 'get-prices-usd', ...coins],
 		queryFn: async () => {
-			const deepBookClient = await getDeepBookClient();
-
 			return getDeepbookPricesInUSD(coins, deepBookClient);
```

Comment:
I think the lint rule is yelling at you because of `deepBookClient`. I think if you just call methods on it vs pass it, it won't complain, just something to factor into your API design